src/qibo/noise.py

```diff
@@ -173,12 +177,50 @@ def add(self, error, gate, qubits=None):
             gate (:class:`qibo.gates.Gate`): gate after which the noise will be added.
             qubits (tuple): qubits where the noise will be applied, if None the noise
                             will be added after every instance of the gate.
+            condition (callable): Optional function that takes :class:`qibo.gates.Gate` object as an input and returns True if noise should be added to it.
+
+        Example:
+
+        .. testcode::
+
+        import numpy as np
+        from qibo import gates
+        from qibo.models import Circuit
+        from qibo.noise import NoiseModel, PauliError
+
+            # Check if a gate is RX(pi/2).
+            def is_sqrt_x(gate):
+                return np.pi/2 in gate.parameters
+
+            # Build a noise model with a Pauli error on RX(pi/2) gates.
+            error = PauliError(0.01, 0.5, 0.1)
+            noise = NoiseModel()
+            noise.add(PauliError(px = 0.5), gates.RX, condition=is_sqrt_x)
+
+            # Generate a noiseless circuit.
+            circuit = Circuit(1)
+            circuit.add(gates.RX(0, np.pi / 2))
+            circuit.add(gates.RX(0, 3 * np.pi / 2))
+            circuit.add(gates.X(0))
+
+            # Apply noise to the circuit.
+            circuit = noise.apply(circuit)
```

Comment:
```suggestion
            noisy_circuit = noise.apply(circuit)
```
I would rename this variable `noisy_circuit` just to avoid losing the original circuit.